### PR TITLE
Adding SI7021 as a model

### PR DESCRIPTION
**Pull request by grea09**
## Description:

Some one wire versions of SI7021 (distributed by Sonoff for their TH modules) requires a different initialization procedure to work. They use the same protocol as the DHT sensors.

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphomeyaml](https://github.com/OttoWinter/esphomeyaml) with YAML changes (if applicable):** OttoWinter/esphomeyaml#375
**Pull request in [esphomelib](https://github.com/OttoWinter/esphomelib) with C++ framework changes (if applicable):** OttoWinter/esphomelib#433

## Checklist:
  - [x] The documentation change has been tested and compiles correctly.
  - [x] Branch: `next` is for changes and new documentation that will go public with the next esphomelib release. Fixes, changes and adjustments for the current release should be created against `current`.

### CODE REVIEW

File: esphomeyaml/components/sensor/dht.rst [hunk 1]
```diff
@@ -62,7 +64,7 @@ Configuration variables:
   - All other options from :ref:`Sensor <config-sensor>` and :ref:`MQTT Component <config-mqtt-component>`.
 
 - **model** (*Optional*, int): Manually specify the DHT model, can be
-  one of ``AUTO_DETECT``, ``DHT11``, ``DHT22``, ``AM2302``, ``RHT03``
+  one of ``AUTO_DETECT``, ``DHT11``, ``DHT22``, ``AM2302``, ``RHT03``, ``SI7021``
```
Comment (OttoWinter) [inline esphomeyaml/components/sensor/dht.rst:67]: Auto detection won't work for this chip. We should make that clear in the docs.

Comment (grea09) [reply esphomeyaml/components/sensor/dht.rst:67]: Does it work for RHT03 and AM2302 ? I couldn't see it in the code.

Comment (OttoWinter) [reply esphomeyaml/components/sensor/dht.rst:67]: Well the RHT03 and AM2302 are just aliases for the DHT22 - so the option is pretty much just there as a placebo :)

**Comment by OttoWinter** [comment]
This PR needs to be against the `next` branch of this repository. Please rebase this commit on top of the `next` branch and change the PR target to next (by clicking on link below title).

**Comment by grea09** [comment]
I'm not an expert on git and branches but I think that will work. I'm sure there's a better way to do the rebase but I don't have time to search for it.

If that's not good I will probably make another PR and close this one.

**Comment by OttoWinter** [comment]
Yes this PR still contains commits that are not from here, that needs to be removed. Easiest way is probably to create new PR, but you can also use this guide: https://developers.home-assistant.io/docs/en/development_catching_up.html

**Comment by grea09** [comment]
I followed the instructions and I think this is good. I am not sure tho.

If it is not good I will close this PR and open another one. I wanted to learn how to do that properly because it keeps happening :smile:

**Comment by OttoWinter** [comment]
No this PR still has other commits on here (see files tab). Opening another PR is probably best option.

**Pull request closed without merging**